db/data/20221114151504_destroy_redundant_irregular_incomes.rb

```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+class DestroyRedundantIrregularIncomes < ActiveRecord::Migration[7.0]
+  def up
+    LegalAidApplication.where(student_finance: false).find_in_batches do |legal_aid_applications|
+      legal_aid_applications.each do |legal_aid_application|
+        legal_aid_application.irregular_incomes.destroy_all
+      end
+    end
+  end
+
+  def down
+    nil
+  end
+end
```

Comment:
Generally I'm wary of using migrations that reference application code, but we've done it lots already.

I used `data_migrate` because it was there, but I don't mind whether we use this, a normal migration, or a temporary rake task.

What do we prefer?

Reply:
Also, are these run automatically on deploy?

Reply:
yes they are run automatically on deploy

Reply:
since this is purely a data cleanup i think it makes sense to have it here

Reply:
currently only 257 applications with student_finance: false so should be quick/fine. 

However, Was wondering if these data migrations implicitly remove the transaction block, as i believe we do not need this, possibly adding our own per batch if needed?


```
disable_ddl_transaction!

def up
    LegalAidApplication.where(student_finance: false).find_in_batches do |legal_aid_applications|
      legal_aid_applications.each do |legal_aid_application|
        legal_aid_application.irregular_incomes.destroy_all
      end
    end
    sleep(0.1) # throttle, but no effect when < 1000
 end

```